File: erezervisi_api/eRezervisi/eRezervisi.Common.Dtos/User/UserCreateDto.cs
namespace eRezervisi.Common.Dtos.User
{
    public class UserCreateDto
    {
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string Phone { get; set; } = null!;
        public string Address { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string? ImageBase64 { get; set; }
        public string? ImageFileName { get; set; }
        public string Username { get; set; } = null!;
        public string Password { get; set; } = null!;
    }
}

File: erezervisi_mobile/erezervisi_mobile/lib/models/requests/user/user_create_dto.dart
class UserCreateDto {
  String firstName;
  String lastName;
  String phone;
  String address;
  String email;
  String username;
  String password;

  UserCreateDto({
    required this.firstName,
    required this.lastName,
    required this.phone,
    required this.address,
    required this.email,
    required this.username,
    required this.password,
  });

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'phone': phone,
      'address': address,
      'email': email,
      'username': username,
      'password': password
    };
  }
}
